### PR TITLE
[FEATURE] - [PXN-2304] - Adding terms to RemedyView if needed

### PR DESCRIPTION
**Pull request by victorpereiradep**
## What have changed?
No mercado crédito por exemplo, exibe os termos de uso acima do botão de pagar, os termos apresentam um texto em destaque e ao clicar nele abre o regulamento.

## Kind of pr:

- [ ] BugFix
- [x] Feature
- [ ] Improvement

### CODE REVIEW

File: MercadoPagoSDK/MercadoPagoSDK/Flows/OneTap/UI/PXOneTapViewController.swift [hunk 1]
```diff
@@ -947,14 +947,9 @@ private extension PXOneTapViewController {
...
-extension PXOneTapViewController: PXTermsAndConditionViewDelegate {
-    func shouldOpenTermsCondition(_ title: String, url: URL) {
-        let webVC = WebViewController(url: url, navigationBarTitle: title)
-        webVC.title = title
-        navigationController?.pushViewController(webVC, animated: true)
-    }
-}
+extension PXOneTapViewController: PXTermsAndConditionViewDelegate { }
 
+// MARK: MLCardFormLifeCycleDelegate
```
Comment (mltheus) [inline MercadoPagoSDK/MercadoPagoSDK/Flows/OneTap/UI/PXOneTapViewController.swift:952]: nice

File: MercadoPagoSDK/MercadoPagoSDK/UI/PXRemedy/PXRemedyView.swift [hunk 2]
```diff
@@ -29,15 +29,69 @@ struct PXRemedyViewData {
     let remedyButtonTapped: ((String?) -> Void)?
 }
 
-class PXRemedyView: UIView {
+final class PXRemedyView: UIView {
+    
+    private lazy var hintLabel: UILabel = {
```
Comment (mltheus) [inline MercadoPagoSDK/MercadoPagoSDK/UI/PXRemedy/PXRemedyView.swift:34]: This way the code went way more organized, nice one 🔥

File: MercadoPagoSDK/MercadoPagoSDK/UI/PXResult/New UI/PXNewResultViewController.swift [hunk 3]
```diff
@@ -804,3 +805,6 @@ extension PXNewResultViewController: ActionViewDelegate {
 
     }
 }
+
+// MARK: PXTermsAndConditionViewDelegate
+extension PXNewResultViewController: PXTermsAndConditionViewDelegate { }
```
Comment (mltheus) [inline MercadoPagoSDK/MercadoPagoSDK/UI/PXResult/New UI/PXNewResultViewController.swift:810]: Why conform with a delegate that has no methods? 🤔 is it supposed to be filled on a next task?

File: MercadoPagoSDK/MercadoPagoSDK/UI/PXReview/PXTermsAndConditionViewDelegate.swift [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+//  Created by Victor Pereira De Paula on 02/09/21.
+//
+
+import Foundation
+
+protocol PXTermsAndConditionViewDelegate: NSObjectProtocol {
+    func shouldOpenTermsCondition(_ title: String, url: URL)
+}
+
+extension PXTermsAndConditionViewDelegate where Self: UIViewController {
```
Comment (mltheus) [inline MercadoPagoSDK/MercadoPagoSDK/UI/PXReview/PXTermsAndConditionViewDelegate.swift:14]: Now i got it, nice approach btw

**Review by mltheus — APPROVED** [review]

File: MercadoPagoSDK/MercadoPagoSDK/UI/PXResult/New UI/PXNewResultViewController.swift (hunk 3 above)
Comment (victorpereiradep) [reply MercadoPagoSDK/MercadoPagoSDK/UI/PXResult/New UI/PXNewResultViewController.swift:810]: Maybe, a better approach is moving PXTermsAndConditionViewDelegate to another extension block or class line

**Review by jscaramal-meli — APPROVED** [review]

**Pull request merged**